fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java

```diff
@@ -162,34 +147,25 @@ protected void onCleared() {
    * message and publishes it to the root channel. It observers the response in the background and
    * switches to the home screen on success.
    */
-  @SuppressLint("CheckResult")
-  public void launchLao() {
+  public void launchLao(Activity activity) {
     String laoName = mLaoName.getValue();
 
     Log.d(TAG, "creating lao with name " + laoName);
     CreateLao createLao = new CreateLao(laoName, keyManager.getMainPublicKey());
+    Lao lao = new Lao(createLao.getId());
 
     disposables.add(
         networkManager
             .getMessageSender()
             .publish(keyManager.getMainKeyPair(), Channel.ROOT, createLao)
+            .doOnComplete(() -> Log.d(TAG, "got success result for create lao"))
+            // Send subscribe and catchup
+            .toObservable()
+            .flatMapCompletable(a -> networkManager.getMessageSender().subscribe(lao.getChannel()))
             .subscribe(
                 () -> {
-                  Log.d(TAG, "got success result for create lao");
-                  Lao lao = new Lao(createLao.getId());
-
-                  // Send subscribe and catchup
-                  networkManager
-                      .getMessageSender()
-                      .subscribe(lao.getChannel())
-                      .subscribe(
-                          () -> {
-                            Log.d(TAG, "subscribing to LAO with id " + lao.getId());
-                            openLAO(lao.getId());
-                          },
-                          error ->
-                              ErrorUtils.logAndShow(
-                                  getApplication(), TAG, error, R.string.error_create_lao));
+                  Log.d(TAG, "subscribing to LAO with id " + lao.getId());
```

Comment:
once this is called, we're already subscribed, right ?

Reply:
Yes, good point

Reply:
Fixed